### PR TITLE
add no-only-test-rule (fixes #5)

### LINKED ISSUE #5: `no-only-test` rule
Same as https://github.com/sindresorhus/eslint-plugin-ava/issues/3, but for `test.only()`.

### PR DESCRIPTION
**Pull request by jfmengels**
add no-only-test-rule (fixes #5) :)

### CODE REVIEW

File: docs/rules/no-only-test.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Ensure no test.only() are present
+
+It's easy to run only one test with test.only() and then forget about it. It's visible in the results, but still easily missed.
```
Comment (jfmengels) [inline docs/rules/no-only-test.md]: Can be improved IMO, but could not figure out a better description.

File: docs/rules/no-only-test.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+
+## Fail
+
+```js
+const test = require('ava');
```
Comment (sindresorhus) [inline docs/rules/no-only-test.md]: Use `import test from 'ava';`

File: docs/rules/no-only-test.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+test.only('some title', t => {
+	t.pass();
+	t.end();
```
Comment (sindresorhus) [inline docs/rules/no-only-test.md]: You can't `t.end()` normal sync tests.

File: docs/rules/no-only-test.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+It's easy to run only one test with test.only() and then forget about it. It's visible in the results, but still easily missed.
+
+
+## Fail
+
+```js
+const test = require('ava');
+
+test.only('some title', t => {
```
Comment (sindresorhus) [inline docs/rules/no-only-test.md]: I would have two tests here with one being `.only` to make it clearer.

File: docs/rules/no-only-test.md (hunk 1 above)
Comment (sindresorhus) [inline docs/rules/no-only-test.md]: Addition

Something like this:

> Forgetting to remove `.only`, means only this one test in the whole file will run, and if not caught, can let serious bugs slip into your codebase.

File: readme.md [hunk 5]
```diff
@@ -45,6 +46,7 @@ Configure it in `package.json`.
 The rules will only activate in test files.
 
 - [no-identical-title](docs/rules/no-identical-title.md) - Ensure no tests have the same title.
+- [no-only-test](docs/rules/no-only-test.md) - Ensure no test.only() are present.
```
Comment (sindresorhus) [inline readme.md]: For consistency, can you wrap "test.only()" in backticks?

File: docs/rules/no-only-test.md (hunk 2 above)
Comment (jfmengels) [inline docs/rules/no-only-test.md]: Sure. Most rules use `require`, so I'll change them all in one go.

**Comment by jfmengels** [comment]
Updated!

File: docs/rules/no-only-test.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
+# Ensure no `test.only()` are present
+
+It's easy to run only one test with `test.only()` and then forget about it. It's visible in the results, but still easily missed. It is potentially disastrous when ran in a CI system as mosts tests will not be run without failing the test suite. Forgetting to remove `.only`, means only this one test in the whole file will run, and if not caught, can let serious bugs slip into your codebase.
```
Comment (sindresorhus) [inline docs/rules/no-only-test.md]: I think you can remove the "It is potentially disastrous when ran in a CI system as mosts tests will not be run without failing the test suite." sentence now that we have the other one.

Comment (jfmengels) [inline docs/rules/no-only-test.md]: :+1:

**Comment by sindresorhus** [comment]
Can you fix the merge conflict?

**Comment by jfmengels** [comment]
Done! (quick question, is it generally useful to tag the reviewers when saying this?)

**Comment by sindresorhus** [comment]
@jfmengels Tagging doesn't really have any effect here though as I'm watching the repo so I get notifications regardless, but if you want to bring someone in that doesn't watch the repo, yeah. Like I tagged @​twada in those other PRs.

**Comment by sindresorhus** [comment]
Excellent work as usual! :)

**Comment by jfmengels** [comment]
Thanks :)

**Pull request merged**